src/components/legend/draw.js

```diff
@@ -769,8 +769,12 @@ function computeLegendDimensions(gd, groups, traces, legendObj) {
     var traceGroupGap = legendObj.tracegroupgap;
     var legendGroupWidths = {};
 
-    var { maxheight, orientation, yref } = legendObj;
-    var heightToBeScaled = orientation === "v" && yref === "paper" ? gs.h : fullLayout.height;
+    const { orientation, yref } = legendObj;
+    let { maxheight } = legendObj;
+    const useFullLayoutHeight = isBelowPlotArea || isAbovePlotArea || !(orientation === "v" && yref === "paper")
```

Comment:
Not to beat a dead horse, but I wonder if this would be clearer?

```suggestion
    const useFullLayoutHeight = isBelowPlotArea || isAbovePlotArea || orientation !== "v" || yref !== "paper"
```

Reply:
That seems fine. I'll update it.